django_su/backends.py

```diff
@@ -13,7 +13,21 @@ class SuBackend(object):
     def authenticate(self, su=False, pk=None, **credentials):
         if not su:
             return None
-        return User.objects.get(pk=pk)
+
+        try:
+            User._meta.pk.get_prep_value(pk)
+        except ValueError:
+            return None
```

Comment:
I'd be willing to accept that the above validation check doesn't need to be done, or could perhaps be done better. This seemed to be the best way I could find though.